velox/functions/sparksql/RegexFunctions.cpp

```diff
@@ -41,6 +30,94 @@ void ensureRegexIsConstant(
   }
 }
 
+/// Spark uses java.util.regex in regexp_replacement. This function preprocesses
+/// the pattern from Spark to ensure it is compatible with RE2.
+/// 1. java.util.regex supports named capturing groups in the format
+/// (?<name>regex), but in RE2, this is written as (?P<name>regex), so we need
+/// to convert the former format to the latter.
+FOLLY_ALWAYS_INLINE std::string prepareRegexpPattern(
```

Comment:
Perhaps move below function to the lib folder for reuse.
https://github.com/facebookincubator/velox/blob/a8bd2dea2f019baea6f25ac9158c76a879bcc668/velox/functions/prestosql/RegexpReplace.h#L33